x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticQueryInnerHitBuilder.java

```diff
@@ -91,7 +91,9 @@ public InnerHitBuilder toInnerHitBuilder() {
 
         return new InnerHitBuilder(fieldName).setFrom(from)
             .setSize(size)
-            .setFetchSourceContext(FetchSourceContext.of(true, null, new String[] { SemanticTextField.getEmbeddingsFieldName(fieldName) }));
+            .setFetchSourceContext(
+                FetchSourceContext.of(true, null, new String[] { SemanticTextField.getEmbeddingsFieldName(fieldName) }, true)
```

Comment:
Will this override vector source filtering being false on inner hits?

Reply:
The idea here is to create a `FetchSourceContext` object for inner hits retrieval that always includes vectors, but filters out the embeddings field. That leaves only the chunk text, which is what we want for inner hits.

Reply:
Might be worth adding a comment here, as that's not really clear.